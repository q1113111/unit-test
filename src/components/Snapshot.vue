<template>
    <a data-test="link" href="http://ithelp.ithome.com.tw"> Issthelaap  </a>
</template>