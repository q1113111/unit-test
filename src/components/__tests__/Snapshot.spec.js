import { mount } from "@vue/test-utils";
import { expect, it } from "vitest";
import Snapshot from '../Snapshot.vue'

// 若有需要預防未預期變更的靜態檔案如「JSON 檔」、「UI 元件」⋯⋯等，都可以透過快照來處理
it('cat snapshot', () => {
    const target = [
        {
            name: 'Orange',
            age: 4
        },
        {
            name: 'Blank',
            age: 6
        }
    ]
    expect(target).toMatchSnapshot()
    //直接產生
     // 原先 toMatchInlineSnapshot 內沒有任何東西
    // 在執行測試後就會將快照內容自動生成在 toMatchInlineSnapshot 裡
    expect(target).toMatchInlineSnapshot(`
      [
        {
          "age": 4,
          "name": "Orange",
        },
        {
          "age": 6,
          "name": "Blank",
        },
      ]
    `)
})

it('snapshot', () => {
    const wrapper = mount(Snapshot)
    const target = wrapper.find('[data-test="link"]')
    // expect(target).toMatchSnapshot()
    expect(target).toMatchInlineSnapshot(`
      DOMWrapper {
        "isDisabled": [Function],
        "wrapperElement": <a
          data-test="link"
          href="http://ithelp.ithome.com.tw"
        >
           Issthelaap 
        </a>,
      }
    `)
})
// 可以引入 JSON 檔案來做快照
// import Area from './area.json'
// it('static json snapshot', () => {
//     expect(Area).toMatchSnapshot()
// })