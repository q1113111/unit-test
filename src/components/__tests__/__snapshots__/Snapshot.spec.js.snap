// Vitest Snapshot v1

exports[`cat snapshot 1`] = `
[
  {
    "age": 4,
    "name": "Orange",
  },
  {
    "age": 6,
    "name": "Blank",
  },
]
`;
